IPython/terminal/prompts.py

```diff
@@ -19,7 +26,7 @@ def in_prompt_tokens(self, cli=None):
 
     def _width(self):
         in_tokens = self.in_prompt_tokens()
-        return sum(len(s) for (t, s) in in_tokens)
+        return sum(len(s) for (t, s) in in_tokens if t is not ZeroWidthEscape)
```

Comment:
no objection, just wanted to know you plan. And thanks for the tip on `token_list_width` !